File: erlang/ch1/ex101.erl
%%% load with > c("ex101.erl").
%%% run with > ex101:expressions().

-module(ex101).
-author("Yohan").

%% API
-export([expressions/0]).

expressions() ->
  io:fwrite("~w\n",[10]),
  io:fwrite("~w\n",[5 + 3 + 4]),
  io:fwrite("~w\n",[9 - 1]),
  io:fwrite("~w\n",[6 / 2]),
  io:fwrite("~w\n",[(2 * 4) + (4 - 6)]),
  A = 3,
  B = A + 1,
  io:fwrite("~w\n",[A + B + (A * B)]),
  io:fwrite("~w\n",[A == B]),
  io:fwrite("~w\n",[if (B > A) and (B < (A * B)) -> B; true -> A end]),
  io:fwrite("~w\n",[if (A == 4) -> 6; (B == 4) -> 6 + 7 + A; true -> 25 end]),
  io:fwrite("~w\n",[2 + (if (B > A) -> B; true -> A end)]),
  io:fwrite("~w\n",[(if (A > B) -> A; (A < B) -> B; true -> -1 end) * (A + 1)]).

File: elixir/ch1/Ex101.ex
# Load with import_file("Ex101.ex")

IO.puts 10

IO.puts 5 + 3 + 4

IO.puts 9 - 1

IO.puts 6 / 2

IO.puts (2*4) + (4-6)

a = 3

b = a + 1

IO.puts a+b+a*b

IO.puts a==b

IO.puts if (b>a && b<(a*b)), do: b, else: a

IO.puts if (a == 4), do: 6, else: if (b == 4), do: 6 + 7 + a, else: 25

IO.puts (if (b > a), do: b, else: a) + 2

IO.puts (if (a > b), do: a, else: if (a < b), do: b, else: -1) * (a + 1)
